### ISSUE TITLE
Expansion context for more rules?

### ISSUE THREAD

**Issue by jberdine** [ocaml-ppx/ppxlib#169]
I am considering a case where `Context_free.Rule.special_function` could be used, if its expand function were given an expansion context similar to `Expansion_context.Extension.t`. Would that, or some other means to obtain the relevant `Code_path.t`, make sense and be reasonably easily implementable?

**Comment by NathanReb** [comment]
Yeah I think we should widen the use of `Expansion_context` to all context free rules indeed. If you'd like to give this a try I'm happy to point you in the right direction, otherwise I'll take a look myself when I get a bit more time!
